src/shared/components/mutationTable/column/ProteinChangeColumnFormatter.tsx

```diff
@@ -95,4 +184,69 @@ export default class ProteinChangeColumnFormatter {
 
         return content;
     }
+
+    private static getProteinChangeData(
+        originalMutation: Mutation,
+        indexedAnnotatedMutationsByGenomicLocation?: {
+            [genomicLocation: string]: Mutation;
+        },
+        isCanonicalTranscript?: boolean
+    ) {
+        // non-canonical
+        if (isCanonicalTranscript === false) {
+            const annotatedMutation = indexedAnnotatedMutationsByGenomicLocation
+                ? findMatchingAnnotatedMutation(
+                      originalMutation,
+                      indexedAnnotatedMutationsByGenomicLocation
+                  )
+                : undefined;
+            if (annotatedMutation) {
+                // non-canonical, GN has data, but data is empty, show empty
+                if (annotatedMutation.proteinChange.length === 0) {
+                    return '';
+                }
+                // non-canonical, GN has data, show GN annotated result
+                return annotatedMutation.proteinChange;
+            }
+            // non-canonical and GN doesn't have data, show data from database
+        }
+        // canonical, show database result
+        return originalMutation.proteinChange;
+    }
+
+    private static shouldShowWarningForDataDifference(
```

Comment:
This function is also a good candidate for some unit tests.

Reply:
we removed the logic here, so no need for tests